### ISSUE TITLE
Improve CPU utilization

### ISSUE THREAD

**Issue by tgraf** [cilium/cilium#5132]
**cilium-agent run 1**
```
      flat  flat%   sum%        cum   cum%
     100ms 14.93% 14.93%      280ms 41.79%  runtime.mallocgc
      80ms 11.94% 26.87%       80ms 11.94%  runtime.heapBitsSetType
      50ms  7.46% 34.33%       90ms 13.43%  runtime.scanobject
      40ms  5.97% 40.30%       40ms  5.97%  runtime.heapBitsForObject
      30ms  4.48% 44.78%       40ms  5.97%  runtime.greyobject
      20ms  2.99% 47.76%      120ms 17.91%  github.com/cilium/cilium/pkg/labels.LabelArray.Get
      20ms  2.99% 50.75%       20ms  2.99%  runtime.(*gcWork).dispose
      20ms  2.99% 53.73%      100ms 14.93%  runtime.gcDrain
      20ms  2.99% 56.72%       30ms  4.48%  runtime.gcmarknewobject
      10ms  1.49% 58.21%       10ms  1.49%  fmt.(*fmt).fmt_s
```

**cilium-agent run 2**
```
      flat  flat%   sum%        cum   cum%
     160ms 45.71% 45.71%      170ms 48.57%  syscall.Syscall
      10ms  2.86% 48.57%       10ms  2.86%  encoding/json.(*encodeState).string
      10ms  2.86% 51.43%      110ms 31.43%  github.com/cilium/cilium/pkg/bpf.(*Map).GetNextKey
      10ms  2.86% 54.29%       10ms  2.86%  github.com/cilium/cilium/pkg/endpoint.(*EndpointStatus).GetModel
      10ms  2.86% 57.14%       10ms  2.86%  github.com/cilium/cilium/pkg/maps/ctmap.(*CtKey4Global).NewValue
      10ms  2.86% 60.00%      230ms 65.71%  github.com/cilium/cilium/pkg/maps/ctmap.doGC4
      10ms  2.86% 62.86%       10ms  2.86%  net/url.shouldEscape
      10ms  2.86% 65.71%       10ms  2.86%  runtime._ExternalCode
      10ms  2.86% 68.57%       20ms  5.71%  runtime.concatstrings
      10ms  2.86% 71.43%       10ms  2.86%  runtime.exitsyscall
```

**cilium-node-monitor run 1**
```
      50ms 17.86% 17.86%       50ms 17.86%  runtime.scanobject
      50ms 17.86% 35.71%       50ms 17.86%  syscall.Syscall
      30ms 10.71% 46.43%       30ms 10.71%  runtime.memmove
      20ms  7.14% 53.57%       20ms  7.14%  runtime.memclrNoHeapPointers
      10ms  3.57% 57.14%       10ms  3.57%  context.(*cancelCtx).Done
      10ms  3.57% 60.71%       10ms  3.57%  encoding/gob.(*Encoder).freeEncoderState
      10ms  3.57% 64.29%      110ms 39.29%  github.com/cilium/cilium/monitor/payload.(*Payload).BuildMessage
      10ms  3.57% 67.86%       10ms  3.57%  reflect.Value.FieldByIndex
      10ms  3.57% 71.43%       30ms 10.71%  runtime.(*mheap).alloc
      10ms  3.57% 75.00%       10ms  3.57%  runtime._ExternalCode
```

**cilium-agent run while deploying pods**
```
      90ms 13.64% 13.64%      160ms 24.24%  runtime.mallocgc
      40ms  6.06% 19.70%       80ms 12.12%  runtime.scanobject
      40ms  6.06% 25.76%       40ms  6.06%  syscall.Syscall
      30ms  4.55% 30.30%       30ms  4.55%  runtime.heapBitsForObject
      30ms  4.55% 34.85%       30ms  4.55%  runtime.indexbytebody
      20ms  3.03% 37.88%       20ms  3.03%  runtime._ExternalCode
      20ms  3.03% 40.91%       40ms  6.06%  runtime.concatstrings
      20ms  3.03% 43.94%       30ms  4.55%  runtime.greyobject
      20ms  3.03% 46.97%       20ms  3.03%  runtime.heapBitsSetType
      10ms  1.52% 48.48%       10ms  1.52%  encoding/base64.(*Encoding).Encode
```

[profiles.zip](https://github.com/cilium/cilium/files/2270680/profiles.zip)

**Comment by tgraf** [comment]
Using monitor API v2:
*cilium-node-monitor*
```
Time: Aug 17, 2018 at 6:32pm (UTC)
Duration: 30.03s, Total samples = 600ms ( 2.00%)
Entering interactive mode (type "help" for commands, "o" for options)
(pprof) top
Showing nodes accounting for 370ms, 61.67% of 600ms total
Showing top 10 nodes out of 132
      flat  flat%   sum%        cum   cum%
     100ms 16.67% 16.67%      110ms 18.33%  syscall.Syscall
      80ms 13.33% 30.00%       80ms 13.33%  runtime.futex
      50ms  8.33% 38.33%       80ms 13.33%  runtime.mallocgc
      30ms  5.00% 43.33%       30ms  5.00%  runtime.epollwait
      30ms  5.00% 48.33%       40ms  6.67%  runtime.scanobject
      20ms  3.33% 51.67%       20ms  3.33%  internal/poll.(*fdMutex).rwunlock
      20ms  3.33% 55.00%       20ms  3.33%  runtime.casgstatus
      20ms  3.33% 58.33%       20ms  3.33%  runtime.exitsyscallfast
      10ms  1.67% 60.00%       10ms  1.67%  encoding/binary.(*encoder).uint32
      10ms  1.67% 61.67%       40ms  6.67%  encoding/gob.(*Encoder).encodeStruct
```

```
Time: Aug 17, 2018 at 6:34pm (UTC)
Duration: 30s, Total samples = 560ms ( 1.87%)
Entering interactive mode (type "help" for commands, "o" for options)
(pprof) top
Showing nodes accounting for 350ms, 62.50% of 560ms total
Showing top 10 nodes out of 120
      flat  flat%   sum%        cum   cum%
     100ms 17.86% 17.86%      100ms 17.86%  runtime.futex
      70ms 12.50% 30.36%       80ms 14.29%  syscall.Syscall
      60ms 10.71% 41.07%       60ms 10.71%  runtime.scanobject
      20ms  3.57% 44.64%       20ms  3.57%  runtime.epollwait
      20ms  3.57% 48.21%       60ms 10.71%  runtime.findrunnable
      20ms  3.57% 51.79%       50ms  8.93%  runtime.mallocgc
      20ms  3.57% 55.36%       20ms  3.57%  runtime.reentersyscall
      20ms  3.57% 58.93%      140ms 25.00%  runtime.schedule
      10ms  1.79% 60.71%       10ms  1.79%  encoding/binary.intDataSize
      10ms  1.79% 62.50%      160ms 28.57%  encoding/gob.(*Encoder).EncodeValue
```
[profile001.svg.zip](https://github.com/cilium/cilium/files/2298327/profile001.svg.zip)

**Comment by aanm** [comment]
This was heavily addressed in v1.5
